File: 285C - Building Permutation.java
import java.util.Collections;
import java.util.Scanner;
import java.util.ArrayList;

public class Solution{
    public static void main( String [] args){
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        ArrayList<Long> v = new ArrayList<Long>();
        for(int i=0;i<n;i++){
            long x = s.nextLong();
            v.add(x);
        }
        long count=0;
        int l=n;
        Collections.sort(v);
        for(int i=0;i<n;i++){
            count+= Math.abs(l- v.get(n-1-i));
            l-=1;
        }
        System.out.println(count);
    }
}

File: 285C - Building Permutation.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;
int main(){
    ll n;
    cin>>n;
    vector <ll> v;
    for(ll i=0,x;i<n;i++){
        cin>>x;
        v.push_back(x);
    }
    sort(v.begin(),v.end());
    ll l=n,count=0;
    for(int i=0;i<n;i++){
        count+= abs(l - v[n-i-1]);
        l-=1;
    }
    cout<<count<<endl;
    return 0;
}
